spec.md

```diff
@@ -202,29 +202,49 @@ help intermediate gateways determine how to route the events.
 
 ### source
 * Type: `URI-reference`
-* Description: This describes the event producer. Often this will include
+* Description: Identifies the event producer. Often this will include
   information such as the type of the event source, the organization
-  publishing the event, the process that produced the event, and some unique
-  identifiers. The exact syntax and semantics behind the data encoded in the URI
-  is event producer defined.
+  publishing the event, the process that produced the event, and some
+  unique identifiers. The syntax of `source` is not defined by this
+  specification.
+
+  In an application with many producers, each producer MUST be
+  assigned a distinct `source`. This specification does not define how
+  to ensure uniqueness. Since `source` is a URI-reference, it allows
+  for standard formats using DNS authorities or UUIDs, as well as
+  application-specific string or path identifiers.
+
 * Constraints:
   * REQUIRED
+
 * Examples
+  * Internet-unique URI with a DNS authority.
     * https://github.com/cloudevents/spec/pull/123
-    * /cloudevents/spec/pull/123
-    * urn:event:from:myapi/resourse/123
     * mailto:cncf-wg-serverless@lists.cncf.io
+  * Universally-unique URN with a UUID:
+    *  urn:uuid:6e8bc430-9c3a-11d9-9669-0800200c9a66
+  * Application-specific path or string identifiers
+    * /cloudevents/spec/pull/123
+    * /sensors/tn-1234567/alerts
+    * 1-555-123-4567
 
 ### id
 * Type: `String`
-* Description: ID of the event. The semantics of this string are explicitly
-  undefined to ease the implementation of producers. Enables deduplication.
+* Description: Identifies of the event, enables de-duplication.  The
+  format of this string is determined by the producer. Each producer
+  MUST generate unique `id` values for its own events, `id` values
+  from different producers might clash. Consumers MAY assume that
```

Comment:
What is the difference between a producer and a source? Is it about unique ìd` values per source?

Reply:
Sorry, I guess you can ignore this comment. I just checked the definition of source. 😊